### ISSUE TITLE
[false-positive] disconnected foreach-statements with Dom/XML classes.

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#37]
```php
$writer = new \XMLWriter();
$writer->openMemory();

foreach ($dispatchNumbers as $number) {
    $writer->startElement('Order');
    $writer->writeAttribute('DispatchNumber', $number);
    $writer->endElement();
}

$writer->endElement();

echo $writer->outputMemory();
```

`startElement` and `endElement` are marked disconnected.

**Comment by funivan** [comment]
Agree. Sometimes Inspection can not detect correct flow

**Comment by funivan** [comment]
One more example: if count <-- inspection fires here
```php
 $visibleFieldsNum = 10;
 $maxVisibleFieldsNum = 15;
 foreach ($allHiddenFields as $fieldUid) {
   if (count($fields) + $visibleFieldsNum >= $maxVisibleFieldsNum) {
     break;
   }

   $fields[] = $fieldUid;
 }
```

**Comment by kalessil** [comment]
@funivan : reproduced

For myself: `$array[] = ...` is not recognized as read/write operation

**Comment by t-kanstantsin** [comment]
It is affected also code with html.
Wiith error:
```php
<?php foreach ($arr as $element): ?>
    <?= $element; ?>
<?php endforeach; ?>
```
WithOUT error:
```php
<?php 
foreach ($arr as $element) {
    echo $element;
}
?>
```

**Comment by kalessil** [comment]
@t-kanstantsin : this has been fixed and needs to be released only.

**Comment by t-kanstantsin** [comment]
ok, thank you

**Comment by kalessil** [comment]
@t-kanstantsin : you are welcome. I'll make a new release within next 48 hours, would you please ping me back if notice this or any new issues?

**Comment by b1rdex** [comment]
@kalessil foreach alternative syntax with html output is fixed now.

**Comment by funivan** [comment]
@kalessil Another false-positive:
```php
<? $user = 'test';?>
<? foreach ([1,2,3] as $index) { ?>
  <?= $user; ?> // Error
<? } ?>
```

P.S. Add test to the plugin and get following result: http://imagizer.imageshack.com/img922/8498/SdWzIe.png

**Comment by rentalhost** [comment]
Seems related to #173, but do not worked with that code.

**Comment by kalessil** [comment]
Closing this ticket: cases have been addressed already.
